tests/test_esm_parser/test_provenance.py

```diff
@@ -2,143 +2,129 @@
     assert config.get_provenance() == check_provenance
 
 
-# Test 5 (reset the provenance of all ``echam`` leaves to "a_string")
-def test_get_provenance_5():
-    config["echam"].set_provenance("a_string")
-    check_provenance = {
-        "echam": {
-            "type": "a_string",
-            "files": {"greenhouse": {"kind": "a_string", "path_in_computer": "a_string"}},
-        },
-        "fesom": {"asd": 2, "model": 2},
-        "computer": None,
-        True: None,
-    }
+# Test 7: Reset the provenance of all ``echam`` leaves to "a_string")
+def test_set_provenance_for_leaf_to_a_string():
+    new_prov = "a_string"
+    config["echam"].set_provenance(new_prov)
+    check_provenance["echam"]["type"] = new_prov
+    check_provenance["echam"]["files"]["greenhouse"]["kind"] = new_prov
+    check_provenance["echam"]["files"]["greenhouse"]["path_in_computer"] = new_prov
+    check_provenance["echam"]["test1"] = new_prov
     assert config.get_provenance() == check_provenance
 
 
-# Test 6 (reset the provenance of a leaf)
-def test_get_provenance_6():
-    config["echam"]["files"]["greenhouse"].provenance["kind"] = "a_new_string"
-    check_provenance = {
-        "echam": {
-            "type": "a_string",
-            "files": {
-                "greenhouse": {"kind": "a_new_string", "path_in_computer": "a_string"}
-            },
-        },
-        "fesom": {"asd": 2, "model": 2},
-        "computer": None,
-        True: None,
-    }
+# Test 8: Rest the provenanve of fesom entry.
+def test_set_provenance_for_a_new_leaf():
+    new_prov = {'line': 2, 'col': 11, 'yaml_file': 'someother.yaml', 'category': 'set_for_onknown_leaf'}
+    config["fesom"] = provenance.DictWithProvenance(config["fesom"], {})
+    config["fesom"].set_provenance(new_prov)
+    check_provenance["fesom"] = {"asd": None}
+    check_provenance["fesom"]["asd"] = new_prov
     assert config.get_provenance() == check_provenance
 
+## Test 9: Reset the provenance of a list")
+#def test_set_provenance_for_a_list_leaf():
+#    new_prov = {'line': 2, 'col': 11, 'yaml_file': 'someother.yaml', 'category': 'this_is_for_a_list'}
+#    config["fesom"] = provenance.DictWithProvenance(config["fesom"], {})
+#    config["fesom"] = {"list": [30, 19]}
+#    config["fesom"]["list"] = provenance.ListWithProvenance(config["fesom"]["list"], [None, None])
+#    config["fesom"]["list"].set_provenance(new_prov)
+#    #print("provenance: ",config["fesom"]["list"].get_provenance())
+#    #print("provenance2: ",config.get_provenance())
+#    assert config.get_provenance() == check_provenance
 
-def test_extract_dict():
+
+# Test 10: Test the extraction of config for all allowed variable types.
+def test_extract_dict_config():
     esm_tools_loader = yaml_to_dict.EsmToolsLoader()
-    file_path = pathlib.Path("example.yaml")
+    os.environ['USER'] = "some_user"
+    file_path = pathlib.Path(str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example.yaml")))
```

Comment:
I think this will break if run locally. What about having a check here whether `os.environ['GITHUB_WORKSPACE']` exists. If it doesn't then you can use the line `file_path = pathlib.Path("example.yaml")`